src/mesh/NodeDB.cpp

```diff
@@ -755,6 +821,171 @@ void NodeDB::saveDeviceStateToDisk()
               &devicestate);
 }
 
+#ifdef USE_PERSISTENT_MSG
+void NodeDB::initSavedMessages()
+{
+    FSCom.mkdir("/msgs");
+
+    updateMessageBounds();
+}
+
+void NodeDB::clearSavedMessages()
+{
+    LOG_DEBUG("Clearing saved messages\n");
+
+    rmDir("/msgs");
+
+    initSavedMessages();
+}
+
+bool NodeDB::messageIsDirectMessage(const meshtastic_MeshPacket &mp)
+{
+    if (mp.to == nodeDB->getNodeNum())
+        return true;
+
+    for (size_t i = 0; i < numMeshNodes; i++)
+        if (mp.to == nodeDB->getMeshNodeByIndex(i)->num)
+            return true;
+
+    return false;
+}
+
+void NodeDB::addMessage(const meshtastic_MeshPacket &mp)
+{
+    auto *msg = new meshtastic_Message();
+    msg->from = mp.from;
```

Comment:
Please use `getFrom()` again here, as packets from the phone originate with `from == 0`.